'let' and 'const' declarations available only when targetting ECMAScript6 and higher error

Hello,

I've recently checked out the project and wanted to build it and run available tests. I've tried this on a FRESH installation of Ubuntu 15.04-desktop Here's the result:

floyd@floyd-VirtualBox:~/blpapi-http$ make
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1203,9): error TS1153: 'let' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1223,11): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1247,9): error TS1153: 'let' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1411,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1412,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1413,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1414,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1686,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1687,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1688,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1689,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1690,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1691,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1692,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1693,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1694,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1695,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1696,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1697,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1698,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1699,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1700,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1701,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1702,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1703,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1704,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1705,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1706,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1707,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1708,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1709,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1710,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1711,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1712,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1713,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1716,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1717,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1718,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1719,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1720,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1721,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1722,15): error TS1154: 'const' declarations are only available when targeting ECMAScript 6 and higher.
/home/floyd/blpapi-http/node_modules/tslint/typings/typescriptServices.d.ts(1757,9): error TS1153: 'let' declarations are only available when targeting ECMAScript 6 and higher.
Makefile:91: recipe for target 'tslint-rules/alignRule.js' failed
make: *** [tslint-rules/alignRule.js] Error 1

I thought something is maybe messed up with my environment, but as I said, it's a fresh installation, I've only installed git and npm. I haven't found a similar issue reported.

Regards,
Tomasz Kupczyk

floyd@floyd-VirtualBox:~$ node --version
v0.10.25

Also, when I try to target ES6 (by changing --target) I get these errors:

floyd@floyd-VirtualBox:~/blpapi-http$ make
tslint-rules/alignRule.ts(81,37): error TS2339: Property 'getLineAndCharacterFromPosition' does not exist on type 'SourceFile'.
tslint-rules/switchDefaultRule.ts(28,35): error TS2339: Property 'clauses' does not exist on type 'SwitchStatement'.
tslint-rules/switchDefaultRule.ts(28,49): error TS7006: Parameter 'clause' implicitly has an 'any' type.
Makefile:91: recipe for target 'tslint-rules/alignRule.js' failed
make: **\* [tslint-rules/alignRule.js] Error 1

I'll try to look into it as well.

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "should": "^5.0.1",
     "socket.io-client": "^1.3.4",
     "tsd": "^0.5.7",
-    "tslint": "^2.1.1",
+    "tslint": "~2.1.1",
     "typescript": "~1.4.1"
   },
   "scripts": {

